src/DI/Compiler.php

```diff
@@ -303,7 +303,11 @@ public static function parseService(ServiceDefinition $definition, $config)
 			$config = array('class' => NULL, 'implement' => $config);
 
 		} elseif ($config instanceof Statement && is_string($config->getEntity()) && interface_exists($config->getEntity())) {
-			$config = array('class' => NULL, 'implement' => $config->getEntity(), 'factory' => array_shift($config->arguments));
+			if (method_exists($config->getEntity(), 'create')) {
```

Comment:
This condition necessary to not break [accessors](https://github.com/nette/di/blob/46a3eccf61ce6de0fa2055473e4fe58cdac9d3de/tests/DI/Compiler.generatedAccessor.phpt#L35-L36) as well.